component/src/ibc/component/client.rs

```
@@ -82,49 +92,45 @@ impl Component for Ics2Client {
 pub(crate) trait Ics2ClientExt: StateWrite {
     // execute a UpdateClient IBC action. this assumes that the UpdateClient has already been
     // validated, including header verification.
-    async fn execute_update_client(&mut self, msg_update_client: &MsgUpdateAnyClient) {
+    async fn execute_update_client(&mut self, msg_update_client: &MsgUpdateClient) {
+        // TODO(erwan): deferred client state deserialization means `execute_update_client` is faillible
+        // see ibc-rs ADR004: https://github.com/cosmos/ibc-rs/blob/main/docs/architecture/adr-004-light-client-crates-extraction.md#light-client-specific-code
+        let tm_header = match msg_update_client.header.type_url.as_str() {
+            TENDERMINT_HEADER_TYPE_URL => {
+                TendermintHeader::try_from(msg_update_client.header.clone())
+                    .expect("decoding tendermint header")
```

Can we bubble up an error here instead, since all execution is fallible now?